### ISSUE TITLE
Error in yaml file are no longer reported

### ISSUE THREAD

**Issue by SzymonMalczak** [asyncapi/vs-asyncapi-preview#200]
#### Describe the bug
Since version 0.6 the errors in the file are no longer reported. 
Before parsing errors were shown in the preview window, now the preview is empty.

#### How to Reproduce

Following file 

asyncapi: 2.6.0
info:
  title: Message
  version: "0.1.0"

Output in previous version:
![image](https://github.com/asyncapi/vs-asyncapi-preview/assets/75781059/c99c7af7-8fe2-4f66-bfe5-7de2843c780b)

Output in new version:
![image](https://github.com/asyncapi/vs-asyncapi-preview/assets/75781059/4466e0c3-cf6f-465b-9270-aa665bdc7191)

#### Expected behavior
Errors should be reported.

**Comment by ivangsa** [comment]
Hi @SzymonMalczak 
thanks for taking the time to report this error, 
it was reported upstream already..

**Comment by ababkin** [comment]
any update on this? is there an upstream link? (i cannot find)
this feature feels pretty important
thanks!

**Comment by catosaurusrex2003** [comment]
@ivangsa 
The current `"@asyncapi/react-component": "^2.2.5"` has got support for showing errors in the document.
Issues:
[https://github.com/asyncapi/asyncapi-react/issues/1048](https://github.com/asyncapi/asyncapi-react/issues/1048)
[https://github.com/asyncapi/asyncapi-react/issues/874](https://github.com/asyncapi/asyncapi-react/issues/874)
The pr which introduces the changes:
[https://github.com/asyncapi/asyncapi-react/pull/1068](https://github.com/asyncapi/asyncapi-react/pull/1068)

But by simply upgrading the `@asyncapi/react-component` dependency in the extension doesnt cause any change in the working of the extension. It still shows a blank page on error. ( i tried this while testing locally )

Am i missing something here which is causes this ?
cuz i think upgrading the dependency should fix the issue of blank screen.

**Comment by ivangsa** [comment]
@catosaurusrex2003  can you have a look at this code:

https://github.com/asyncapi/vs-asyncapi-preview/blob/master/src/PreviewWebPanel.ts#L142

maybe now we need to instantiate this with different parameters

**Comment by catosaurusrex2003** [comment]
@ivangsa i found what the issue is.

only the `schema.options` needs to be changed to `schema.requestOptions`

But even doing this wont make errors render.

the problem is present in [https://github.com/asyncapi/asyncapi-react/blob/master/library/src/containers/AsyncApi/Standalone.tsx](https://github.com/asyncapi/asyncapi-react/blob/master/library/src/containers/AsyncApi/Standalone.tsx)

The problem is that `AsyncApiComponent` inside `Standalone.tsx` relies on the error props being passed down onto it from `AsyncApiComponent` inside `AsyncApi.tsx`.
( Hence why errors are rendered in the playground and not in the extension. cuz we are directly using standalone without any parent )

This might have been my short-sightedness while doing the PR.

So the solution for this is to make it so that `SpecificationHelpers.retrieveParsedSpec` supports returning errors so that `<Error />` can be conditionally rendered even without the need for the props ( This will make it work in our extension ) 

Pls correct me if i am wrong or if i am missing ourt on the bigger picture.

**Comment by ivangsa** [comment]
@catosaurusrex2003 you know more than me about this

propose what do you think is the best solution,

can we pass any kind of object to `AsyncApiStandalone.render` that will held the errors in case there are any?

**Comment by catosaurusrex2003** [comment]
> can we pass any kind of object to AsyncApiStandalone.render that will held the errors in case there are any?

I dont think this will be possible because this will require us to parse the document so that we can pass errors to `AsyncApiStandalone.render`. This defeats the purpose of having a `AsyncApiStandalone` in itself.

> So the solution for this is to make it so that SpecificationHelpers.retrieveParsedSpec supports returning errors so that <Error /> can be conditionally rendered even without the need for the props ( This will make it work in our extension )

I think this is the only way to go as of now.

Let me work on modifying `AsyncApiStandalone` and test the playground and this extension both locally and raise a PR for this.
I will get back to you @ivangsa.

@AceTheCreator you are more well versed with `@asyncapi/react-component`. Please let me know if i am on the right track.

**Comment by catosaurusrex2003** [comment]
@ivangsa 

![image](https://github.com/user-attachments/assets/3f1e93df-cadb-4329-9b06-f11d2947cd38)

needed to do a lot of changes in `@asyncapi/react-component` and now working.
Will create a PR in that repo and hopefully it will get merged and start working here after we bump the version.

**Comment by catosaurusrex2003** [comment]
![image](https://github.com/user-attachments/assets/a755ca23-181c-450d-8e25-265b08f39996)

Update: The issue has been fixed in v0.6.5.

There is still a minor CSS issue with the gray text. I'll investigate further to pinpoint the exact problem and work on a fix.

**Comment by SURAJ-RATHI** [comment]
I think this issue has been resolved  i.e it seems perfect on my side here is the screenshot:

![Image](https://github.com/user-attachments/assets/8197c385-62f9-4375-a8aa-c79260d28d29)
